base/hamt.jl

```diff
@@ -75,6 +75,9 @@ function HAMT{K,V}(k::K, v) where {K, V}
 end
 HAMT(k::K, v::V) where {K, V} = HAMT{K,V}(K, V)
 
+hash(x) = objectid(x)::UInt
+hash(x, h::UInt) = hash(x) - 3h
```

Comment:
```suggestion
hash(x, h::UInt) = hash(x) - 3h
```

I think it would make sense to do `s/hash(x)/objectid(x)/`?